### PR TITLE
Add glossary feature

### PR DESCRIPTION
**Pull request by zubairnahmed**
Fixes: #300 

- adds /glossary route
- adds /glossary/:termSlugs route
- add add-term script

### CODE REVIEW

File: web-server/routes/glossary.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const url = require('url')
+const bodyParser = require('body-parser')
+const queries = require('../../database/queries')
+const commands = require('../../database/commands')
+
+module.exports = app => {
+
```
Comment (deadlyicon) [inline web-server/routes/glossary.js]: I think we want it for all glossary routes `app.use('/glossary', app.ensureTrailingSlash)`

File: web-server/routes/glossary.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  app.get('/glossary', app.ensureTrailingSlash, (request, response, next) => {
+    const userId = request.user.id
+    const termSlugs = response.digest.glossary
+    response.render('glossary/index', {
+      termSlugs,
+      title: 'Glossary Terms',
+    })
+  })
+
+  app.get('/glossary/:termSlug', app.ensureTrailingSlash)
```
Comment (deadlyicon) [inline web-server/routes/glossary.js]: if you do above you can remove this

File: digest/glossary.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  utils.readdir('/glossary')
+    .then(convertGlossaryFileNamesToTerms)
+    .then(indexById)
+
+const convertGlossaryFileNamesToTerms = glossaryFileName =>
+  glossaryFileName
+    .map(fileName => ({
+      fileName,
+      id: fileName,
+      value: fileName.replace(/-/g, ' '),
```
Comment (deadlyicon) [inline digest/glossary.js]: let's extract it from the top level heading of the markdown file. This means you'll need to create a `Promise.all` if terms mapped to a function that reads the `README.md` file, uses the markdown lexer to parse it, extracts the level one head text value and puts it on the term under the `value` property.

File: digest/glossary.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      path: `/glossary/${fileName}`,
+      markdownFilePath: `/glossary/${fileName}/README.md`,
```
Comment (deadlyicon) [inline digest/glossary.js]: were not using this property. lets remove it

File: digest/glossary.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    }))
+
+const indexById = terms => {
```
Comment (deadlyicon) [inline digest/glossary.js]: dont we have a helper for this?

File: web-server/routes/glossary.js (hunk 1 above)
Comment (zubairnahmed) [reply web-server/routes/glossary.js]: This isn't working. We followed the pattern we found in modules.js. We left it as is for further comments.

File: web-server/routes/glossary.js (hunk 2 above)
Comment (zubairnahmed) [reply web-server/routes/glossary.js]: Please see above.

File: digest/glossary.js (hunk 4 above)
Comment (zubairnahmed) [reply digest/glossary.js]: We're using the 'path' property in our views. 

'markdownFilePath' is needed per spec. Should we remove it?

File: digest/glossary.js (hunk 5 above)
Comment (zubairnahmed) [reply digest/glossary.js]: We're following the pattern we found in modules.js. It doesn't appear that we have a helper function this.

File: digest/glossary.js (hunk 3 above)
Comment (hhhhhaaaa) [reply digest/glossary.js]: So, to clarify, we are establishing value's property by calling one helper (readMarkdownFile) to read the markdown file, and then call another helper (extractListFromSection) to extract the title from the markdown file?

**Pull request closed without merging**